app/services/mapping_migration_service.rb

```diff
@@ -0,0 +1,100 @@
+class MappingMigrationService
+  LEGACY_FIELDS_REGEX = /%(\S+?)%/
+  attr_reader :integration_name, :rtp_id
+
+  def call
+    upload_integrations = Dradis::Plugins::with_feature(:upload)
+    upload_integrations = upload_integrations - [
+      Dradis::Plugins::Projects::Engine, Dradis::Plugins::CSV::Engine
+    ]
+
+    upload_integrations.each do |integration|
+      @integration_name = integration.plugin_name.to_s
+
+      if integration.uploaders.count > 1
+        migrate_multiple_upload_integration(integration)
+      else
+        integration_template_files.each do |template_file|
+          mapping_source = File.basename(template_file, '.template')
+          # create a mapping & mapping_fields for each field in the file
+          migrate(template_file, mapping_source)
+        end
+      end
+    end
+  end
+
+  private
+
+  def create_mapping(mapping_source)
+    destination = rtp_id ? "rtp_#{rtp_id}" : nil
+
+    Mapping.find_or_create_by!(
+      component: integration_name,
+      source: mapping_source,
+      destination: destination
+    )
+  end
+
+  def create_mapping_fields(mapping, template_file)
+    template_fields = parse_template_fields(template_file)
+
+    # create a mapping_field for each field in the .template file
+    template_fields.each do |field_title, field_content|
+      # set source_field by taking the first match to the existing %% syntax
+      source_field = field_content.match(LEGACY_FIELDS_REGEX)
+      source_field = source_field ? source_field[1] : 'custom text'
+
+      updated_content = update_syntax(field_content)
+
+      mapping.mapping_fields.find_or_create_by!(
+        source_field: source_field,
+        destination_field: field_title,
+        content: updated_content
+      )
+    end
+  end
+
+  def migrate(template_file, mapping_source)
+    rtp_ids = defined?(Dradis::Pro) ? ReportTemplateProperties.ids : [nil]
+    rtp_ids.each do |rtp_id|
+      @rtp_id = rtp_id
+
+      ActiveRecord::Base.transaction do
+        mapping = create_mapping(mapping_source)
+        create_mapping_fields(mapping, template_file)
+        File.rename template_file, "#{template_file}.legacy"
+      end
+    end
+  end
+
+  def migrate_multiple_upload_integration(integration)
```

Comment:
Can you add a comment why we need this method? With specific template name examples?